src/shared/components/resources/ResourceTab.tsx

```diff
@@ -93,6 +93,14 @@ export default class ResourceTab extends React.Component<
                 url = buildPDFUrl(this.currentResourceDatum.url);
                 break;
         }
+
+        // apply instance specific transformation on url (e.g. to keep state)
+        CUSTOM_URL_TRANSFORMERS.forEach(config => {
+            if (config.re.test(getServerConfig().app_name || '')) {
+                url = config.func(url);
+            }
+        });
```

Comment:
maybe it can be applied for all apps, but we only apply it for those resources where any one of these fields contains the string 'minerva':

```
description: "Minerva story"
displayName: "Minerva Story"
resourceId: "MINERVA_STORY"
```